### PR TITLE
vertical icon fixes

### PR DESCRIPTION
**Pull request by bunsenstraat**
should fix:
- [x] activating home with logo
- [x] activating filepanel when closing an active plugin in icons
- [x] scrolling the icons - sort of scrolling....
- [x] fixed doc link not working
- [x] removed unneeded classes for icon types
- [x] refactor icon rendering and sorting
- [x] rm deps on constructors

### CODE REVIEW

**Comment by LianaHus** [comment]
issue: i don't see top/bottom borders and chevrons after reload, I need to add a plugin to see them

File: apps/remix-ide/src/app/components/vertical-icons.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+const profile = {
+  name: 'menuicons',
+  displayName: 'Vertical Icons',
+  description: '',
+  version: packageJson.version,
+  methods: ['select', 'unlinkContent', 'linkContent'],
+  events: ['toggleContent', 'showContent']
+}
+
+// TODO merge with side-panel.js. VerticalIcons should not be a plugin
```
Comment (LianaHus) [inline apps/remix-ide/src/app/components/vertical-icons.tsx:21]: please create an issue for todo and add a number here

File: apps/remix-ide/src/app/components/side-panel.tsx [hunk 2]
```diff
@@ -19,18 +18,17 @@ const sidePanel = {
 
 // TODO merge with vertical-icons.js
```
Comment (LianaHus) [inline apps/remix-ide/src/app/components/side-panel.tsx:19]: export class VerticalIcons extends Plugin {

Comment (bunsenstraat) [reply apps/remix-ide/src/app/components/side-panel.tsx:19]: what do you mean?

File: apps/remix-ide/src/app/components/vertical-icons.tsx (hunk 1 above)
Comment (bunsenstraat) [reply apps/remix-ide/src/app/components/vertical-icons.tsx:21]: ah old comment can be removed

File: apps/remix-ide/src/app/components/side-panel.tsx (hunk 2 above)
Comment (bunsenstraat) [reply apps/remix-ide/src/app/components/side-panel.tsx:19]: ah not true, should be a plugin I removed that old comment

Comment (LianaHus) [reply apps/remix-ide/src/app/components/side-panel.tsx]: :)) I 'pasted' wrong line. comment was about todo

**Review by LianaHus — APPROVED** [review]

**Pull request merged**